src/backend/app/Jobs/ProspectExtract.php

```diff
@@ -0,0 +1,253 @@
+                'serial_number' => $deviceData->serial_number ?? '',
+                'site_name' => $primaryAddress->street ?? null,
+
+                // Payment plan information
+                'payment_plan_amount_financed_principal' => null,
+                'payment_plan_amount_financed_interest' => null,
+                'payment_plan_amount_financed_total' => null,
+                'payment_plan_amount_down_payment' => $assetPerson->down_payment ?? null,
+                'payment_plan_cash_price' => $assetPerson->total_cost ?? null,
+                'payment_plan_currency' => null,
+                'payment_plan_installment_amount' => null,
+                'payment_plan_number_of_installments' => $assetPerson->rate_count ?? null,
+                'payment_plan_installment_period_days' => null,
+                'payment_plan_days_financed' => null,
+                'payment_plan_days_down_payment' => null,
+                'payment_plan_category' => 'paygo',
+
+                // Dates
+                'purchase_date' => $device->created_at->format('Y-m-d'),
+                'installation_date' => $device->created_at->format('Y-m-d'),
+                'repossession_date' => null,
+                'paid_off_date' => null,
+                'repossession_category' => null,
+                'write_off_date' => null,
+                'write_off_reason' => null,
+
+                'is_test' => false,
+                'latitude' => $latitude,
+                'longitude' => $longitude,
+                'country' => $primaryAddress->city->country->country_code ?? null,
+                'location_area_1' => $primaryAddress->city->country->country_name ?? null,
+                'location_area_2' => $primaryAddress->city->name ?? null,
+                'location_area_3' => null,
+                'location_area_4' => null,
+                'location_area_5' => null,
+            ];
+
+            $installations[] = $installation;
+        }
+
+        $count = count($installations);
+        Log::info('Loaded '.$count.' '.($count === 1 ? 'installation' : 'installations').' from database');
+
+        return $installations;
+    }
+
+    /**
+     * Generate filename for CSV.
+     *
+     * @return string
+     */
+    private function generateFileName(): string {
+        $timestamp = now()->toISOString();
+
+        return "prospect_{$timestamp}.csv";
+    }
+
+    /**
+     * Write data to CSV file.
+     *
+     * @param array<int, array<string, mixed>> $data
+     * @param string                           $fileName
+     *
+     * @return string
+     */
+    private function writeCsvFile(array $data, string $fileName): string {
+        $headers = array_keys($data[0]);
+        $csvContent = $this->arrayToCsv($data, $headers);
+
+        // Get company database from the current context
+        $companyDatabase = app(\App\Models\CompanyDatabase::class)->newQuery()->first();
+        $companyDatabaseName = $companyDatabase->getDatabaseName();
```

Comment:
Please use the service layer as much as possible; referencing the models directly does form tight coupling. 

```php
 
  $companyDatabase = app(CompanyDatabaseService::class)->findByCompanyId($this->companyId);
  $companyDatabaseName = $companyDatabase->getDatabaseName();
```